subprojects/testing-jvm-infrastructure/src/main/java/org/gradle/api/internal/tasks/testing/testng/TestNGTestClassProcessor.java

```diff
@@ -74,6 +74,9 @@ public void startProcessing(TestResultProcessor resultProcessor) {
         resultProcessorActor = actorFactory.createBlockingActor(resultProcessor);
         this.resultProcessor = resultProcessorActor.getProxy(TestResultProcessor.class);
         applicationClassLoader = Thread.currentThread().getContextClassLoader();
+        if (spec.isDryRun()) {
+            System.setProperty("testng.mode.dryrun", "true");
```

Comment:
Does this system property ever get reset? What happens if another testng Test task executes on test worker, without dry run, after this one is done executing?

Reply:
TestNG reverting this property after test run:
https://github.com/testng-team/testng/blob/1288058fac43aa200ee2a89f3a779e5604ca7011/testng-core/src/test/java/org/testng/DryRunTest.java#L20
but I've added a test case as well:
https://github.com/gradle/gradle/pull/22308/commits/a748b4e2cc07555d4a83d800e01281178c068954#diff-a9ebdb98aec251217c7a9b48c898d0f195d4bf7711c01083841367e44f4ccf35R38

Reply:
Also, do we reuse test daemons between gradle invocations, even if we reuse the gradle daemon?

I'm worried that two tests within the same invocation of Gradle which execute two separate Test tasks may conflict here

Reply:
Yeah, I'm understanding your concern. I'm not sure about reusing test daemons thought. Can somebody from JVM team answer on it?

Reply:
I'm trying to figure out if we keep them alive across builds, but I am having trouble. @ghale do you know?

In any case, I think we should be able to test this relatively easily. In one build, create two `Test` tasks, or better yet use test suites to do so. The first should have dry run enabled. Then, ensure the second `Test` task `mustRunAfter` the first. Then, also ensure `maxParallelForks` = 1 (though this should default to 1)

Reply:
@jvandort PTAL at the last commit, did I'm correctly understood what you meant?

Reply:
Yeah this was what I was looking for. Though I am surprised this works. We must save system properties somewhere and restore them after a test suite runs. After thinking about it we must have needed to deal with this other places already. 

Thanks for adding the test